File: ios/SsdidWallet/Feature/Credentials/CredentialsScreen.swift
import SwiftUI

struct CredentialsScreen: View {
    @Environment(AppRouter.self) private var router

    @State private var credentials: [VerifiableCredential] = []

    var body: some View {
        VStack(spacing: 0) {
            // Header
            HStack(spacing: 4) {
                Button { router.pop() } label: {
                    Image(systemName: "chevron.left")
                        .foregroundStyle(Color.textPrimary)
                        .font(.system(size: 20))
                }
                .padding(.leading, 8)

                Text("Credentials")
                    .font(.ssdidHeadline)
                    .foregroundStyle(Color.textPrimary)

                Spacer()
            }
            .padding(.vertical, 12)
            .padding(.trailing, 20)

            ScrollView {
                LazyVStack(spacing: 10) {
                    ForEach(credentials) { vc in
                        Button {
                            router.push(.credentialDetail(credentialId: vc.id))
                        } label: {
                            HStack(spacing: 0) {
                                Rectangle()
                                    .fill(Color.ssdidAccent)
                                    .frame(width: 4)

                                VStack(alignment: .leading, spacing: 0) {
                                    let isExpired = isCredentialExpired(vc)

                                    HStack {
                                        Text(vc.type.last ?? "Credential")
                                            .font(.ssdidCaption)
                                            .foregroundStyle(Color.textTertiary)
                                        Spacer()
                                        Text(isExpired ? "Expired" : "Valid")
                                            .font(.system(size: 10))
                                            .foregroundStyle(isExpired ? Color.danger : Color.success)
                                            .padding(.horizontal, 8)
                                            .padding(.vertical, 2)
                                            .background(isExpired ? Color.dangerDim : Color.successDim)
                                            .cornerRadius(4)
                                    }

                                    Spacer().frame(height: 4)

                                    Text(String(vc.id.suffix(20)))
                                        .font(.ssdidHeadline)
                                        .foregroundStyle(Color.textPrimary)

                                    Spacer().frame(height: 4)

                                    Text("Issuer: \(vc.issuer.truncatedDid)")
                                        .font(.system(size: 11, design: .monospaced))
                                        .foregroundStyle(Color.textTertiary)
                                        .lineLimit(1)

                                    Spacer().frame(height: 8)

                                    HStack {
                                        Text("Issued: \(String(vc.issuanceDate.prefix(10)))")
                                            .font(.system(size: 12))
                                            .foregroundStyle(Color.textSecondary)
                                        Spacer()
                                        if let exp = vc.expirationDate {
                                            Text("Expires: \(String(exp.prefix(10)))")
                                                .font(.system(size: 12))
                                                .foregroundStyle(Color.textSecondary)
                                        }
                                    }
                                }
                                .padding(14)
                            }
                            .background(Color.bgCard)
                            .cornerRadius(16)
                        }
                    }

                    if credentials.isEmpty {
                        VStack(spacing: 0) {
                            ZStack {
                                RoundedRectangle(cornerRadius: 20)
                                    .fill(Color.accentDim)
                                    .frame(width: 72, height: 72)
                                Image(systemName: "doc.text")
                                    .font(.system(size: 32))
                                    .foregroundStyle(Color.ssdidAccent)
                            }
                            Spacer().frame(height: 16)
                            Text("No credentials yet")
                                .font(.ssdidHeadline)
                                .foregroundStyle(Color.textPrimary)
                            Spacer().frame(height: 4)
                            Text("Register with a service to receive your first credential")
                                .font(.system(size: 14))
                                .foregroundStyle(Color.textSecondary)
                                .multilineTextAlignment(.center)
                        }
                        .frame(maxWidth: .infinity)
                        .padding(32)
                        .background(Color.bgCard)
                        .cornerRadius(16)
                    }
                }
                .padding(.horizontal, 20)
            }
        }
        .background(Color.bgPrimary)
    }

    private func isCredentialExpired(_ vc: VerifiableCredential) -> Bool {
        guard let exp = vc.expirationDate else { return false }
        let formatter = ISO8601DateFormatter()
        guard let date = formatter.date(from: exp) else { return false }
        return Date() > date
    }
}

File: android/app/src/main/java/my/ssdid/wallet/feature/credentials/CredentialsScreen.kt
package my.ssdid.wallet.feature.credentials

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import my.ssdid.wallet.domain.model.VerifiableCredential
import my.ssdid.wallet.domain.vault.Vault
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import my.ssdid.wallet.ui.components.truncatedDid
import my.ssdid.wallet.ui.theme.*
import java.time.Instant
import javax.inject.Inject

@HiltViewModel
class CredentialsViewModel @Inject constructor(private val vault: Vault) : ViewModel() {
    private val _credentials = MutableStateFlow<List<VerifiableCredential>>(emptyList())
    val credentials = _credentials.asStateFlow()

    init { viewModelScope.launch { _credentials.value = vault.listCredentials() } }
}

@Composable
fun CredentialsScreen(
    onBack: () -> Unit,
    onCredentialClick: (String) -> Unit,
    viewModel: CredentialsViewModel = hiltViewModel()
) {
    val credentials by viewModel.credentials.collectAsState()

    Column(Modifier.fillMaxSize().background(BgPrimary).statusBarsPadding()) {
        Row(
            Modifier.padding(start = 8.dp, end = 20.dp, top = 12.dp, bottom = 12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            IconButton(onClick = onBack) {
                Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back", tint = TextPrimary)
            }
            Spacer(Modifier.width(4.dp))
            Text("Credentials", style = MaterialTheme.typography.titleLarge)
        }

        LazyColumn(
            Modifier.padding(horizontal = 20.dp),
            verticalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            items(credentials) { vc ->
                val credType = vc.type.lastOrNull() ?: "Credential"
                Card(
                    Modifier
                        .fillMaxWidth()
                        .clickable { onCredentialClick(vc.id) }
                        .semantics { contentDescription = "$credType from ${vc.issuer}" },
                    shape = RoundedCornerShape(16.dp),
                    colors = CardDefaults.cardColors(containerColor = BgCard)
                ) {
                    Row {
                        Box(Modifier.width(4.dp).height(100.dp).background(Accent))
                        Column(Modifier.padding(14.dp)) {
                            val isExpired = vc.expirationDate?.let {
                                try { Instant.now().isAfter(Instant.parse(it)) } catch (_: Exception) { false }
                            } ?: false
                            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                                Text(vc.type.lastOrNull() ?: "Credential", style = MaterialTheme.typography.labelSmall)
                                Box(Modifier.clip(RoundedCornerShape(4.dp)).background(if (isExpired) DangerDim else SuccessDim).padding(horizontal = 8.dp, vertical = 2.dp)) {
                                    Text(if (isExpired) "Expired" else "Valid", fontSize = 10.sp, color = if (isExpired) Danger else Success)
                                }
                            }
                            Spacer(Modifier.height(4.dp))
                            Text(vc.id.truncatedDid(), style = MaterialTheme.typography.headlineSmall)
                            Spacer(Modifier.height(4.dp))
                            Text("Issuer: ${vc.issuer}", fontFamily = FontFamily.Monospace, fontSize = 11.sp, color = TextTertiary, maxLines = 1)
                            Spacer(Modifier.height(8.dp))
                            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                                Text("Issued: ${vc.issuanceDate.take(10)}", style = MaterialTheme.typography.bodySmall)
                                vc.expirationDate?.let { Text("Expires: ${it.take(10)}", style = MaterialTheme.typography.bodySmall) }
                            }
                        }
                    }
                }
            }

            if (credentials.isEmpty()) {
                item {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(16.dp))
                            .background(BgCard)
                            .padding(32.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Box(
                            modifier = Modifier
                                .size(64.dp)
                                .clip(CircleShape)
                                .background(AccentDim),
                            contentAlignment = Alignment.Center
                        ) {
                            Text("\uD83D\uDCC4", fontSize = 28.sp)
                        }
                        Spacer(Modifier.height(16.dp))
                        Text(
                            "No credentials yet",
                            style = MaterialTheme.typography.headlineSmall,
                            color = TextPrimary
                        )
                        Spacer(Modifier.height(8.dp))
                        Text(
                            "Register with a service to receive your first credential",
                            style = MaterialTheme.typography.bodyMedium,
                            color = TextSecondary
                        )
                    }
                }
            }
        }
    }
}
